server/app.js

```diff
@@ -8,26 +8,32 @@ const createTest = require('./routes/createTest');
 const profileStats = require('./routes/profileStats');
 const authRoutes = require('./routes/authRoutes');
 const passport = require('passport');
-const cookieSession = require('cookie-session');
 const app = express();
-require('./controllers/AuthController');
+require('./passport-setup');
 
 const port = process.env.PORT;
+const authenticate = passport.authenticate('jwt', { session: false });
 
 app.use(morgan('combined'));
 app.use(bodyParser.json());
 
-app.use(cookieSession({
-  maxAge: 24 * 60 * 60 * 1000,
-  keys: [process.env.COOKIE_KEY]
-}));
-
 app.use(passport.initialize());
-app.use(passport.session());
 
-app.use('/auth', authRoutes);
-app.use('/profile', profileStats);
-app.use('/topic', createTest);
+app.use('/auth', authRoutes, handleError);
+app.use('/profile', authenticate, profileStats);
+app.use('/topic', authenticate, createTest);
+
+function handleError(err, req, res, next) {
+  if (err.name === 'AuthenticationError') {
+    res.status(401).send({
+      message: 'User already exists.'
+    });
+  } else if (err.name === 'SequelizeValidationError') {
+    res.status(401).send({
+      message: 'Invalid input form.'
+    });
+  }
```

Comment:
Since you are using the same status code, diff message:

```js
let message = '';
if (err.name === 'AuthenticationError') message = 'User already exists.'
if (err.name === 'SequelizeValidationError') message = 'Invalid input form.'
res.status(401).send({ message });
```

Might as well use Http status codes package... (optional) 😃